apis/cluster/v1alpha1/types.go

```diff
@@ -24,9 +24,11 @@ import (
 
 // ClusterParameters define the desired state of an ArgoCD Cluster
 type ClusterParameters struct {
-	// Server is the API server URL of the Kubernetes cluster
+	// Server is the API server URL of the Kubernetes cluster. Optional if using a kubeconfig
+	// +optional
 	Server string `json:"server"`
```

Comment:
Optional fields should be declared as pointers.

Reply:
I've updated the PR to make this field a pointer.